sdk/resourcemanager/azure-resourcemanager-containerservice/src/main/java/com/azure/resourcemanager/containerservice/implementation/KubernetesClusterImpl.java

```diff
@@ -436,6 +445,31 @@ private void clearKubeConfig() {
         this.formatUserKubeConfigsMap.clear();
     }
 
+    @Override
+    public KubernetesClusterImpl withFreeSku() {
+        this.innerModel().withSku(new ManagedClusterSku().withTier(ManagedClusterSkuTier.FREE).withName(ManagedClusterSkuName.BASE));
+        if (!KubernetesSupportPlan.KUBERNETES_OFFICIAL.equals(this.innerModel().supportPlan())) {
+            this.innerModel().withSupportPlan(KubernetesSupportPlan.KUBERNETES_OFFICIAL);
+        }
```

Comment:
I mean, do we need to set the supportPlan?

What if we don't set that at all for Free/Standard?

Reply:
Yes, it need to set the supportPlan. The `AKSLONG_TERM_SUPPORT` only applies to `PREMIUM`.

Reply:
What do you mean? If the `supportPlan=AKSLONG_TERM_SUPPORT`, request will fail?

This seems not same as I heard in yesterday's meeting? @XiaofeiCao

Reply:
![image](https://github.com/Azure/azure-sdk-for-java/assets/74638143/e4b7e7de-dfa2-4626-a38e-bb5f042fa380)

Reply:
paste me a log of your request JSON and response JSON

Reply:
Default settings of Kubernetes Cluster:
Sku: FREE
SupportPlan: KUBERNETES_OFFICIA

`kubernetesCluster.update().withFreeSku(KubernetesSupportPlan.AKSLONG_TERM_SUPPORT).apply();`
[test.log](https://github.com/Azure/azure-sdk-for-java/files/14519405/test.log)

`kubernetesCluster.update().withStandardSku(KubernetesSupportPlan.AKSLONG_TERM_SUPPORT).apply();`
[test.log](https://github.com/Azure/azure-sdk-for-java/files/14519509/test.log)

Reply:
OK. Then just delete the `if (!KubernetesSupportPlan.KUBERNETES_OFFICIAL.equals(this.innerModel().supportPlan())) {`.

Reply:
Strange. I wonder what [this](https://learn.microsoft.com/en-us/azure/aks/long-term-support#enable-long-term-support) means:
<img width="852" alt="Screenshot 2024-03-07 at 2 46 35 PM" src="https://github.com/Azure/azure-sdk-for-java/assets/92354331/30c68c33-e7a0-4d10-910a-c52a1f3e3001">

If they are bound together, we can set them for the user I think.